test/neuron/connection_test.exs

```diff
@@ -0,0 +1,30 @@
+defmodule Neuron.ConnectionTest do
+  use ExUnit.Case
+
+  alias Neuron.Connection
+
+  describe "build_headers/0" do
+    setup do
+      Neuron.Config.set(nil)
+    end
+
+    test "sets content-type as default header" do
+      assert Connection.build_headers() == ["Content-Type": "application/graphql"]
```

Comment:
This tests are failing because `build_headers/0` is a private function (as it should be). Maybe we could instead check that `post/2` calls `HTTPotion.post/4` with correct arguments (which would actually also include the test coverage 🎉).